### ISSUE TITLE
Invalid license issue

### ISSUE THREAD

**Issue by iahmad-khan** [floragunncom/search-guard#490]
I am facing a strange issue here , after I uploaded the key to searchguard through kibana interface:

1. when i run sgadmin.sh , all the configuration is deleted
2. i have to put the license again through kibana interface , othervise it defaults to trial

```
docker exec 80 bin/init_sg.sh

Search Guard Admin v6
Will connect to localhost:9300 ... done
Elasticsearch Version: 6.2.1
Search Guard Version: 6.2.1-22.0
Connected as CN=kirk,OU=client,O=client,L=Test,C=DE
Contacting elasticsearch cluster 'elasticsearch' and wait for YELLOW clusterstate ...
Clustername: docker-cluster
Clusterstate: GREEN
Number of nodes: 5
Number of data nodes: 5
searchguard index already exists, so we do not need to create one.
Populate config from /usr/share/elasticsearch/config/sg
Will update 'sg/config' with config/sg/sg_config.yml 
   SUCC: Configuration for 'config' created or updated
Will update 'sg/roles' with config/sg/sg_roles.yml 
   SUCC: Configuration for 'roles' created or updated
Will update 'sg/rolesmapping' with config/sg/sg_roles_mapping.yml 
   SUCC: Configuration for 'rolesmapping' created or updated
Will update 'sg/internalusers' with config/sg/sg_internal_users.yml 
   SUCC: Configuration for 'internalusers' created or updated
Will update 'sg/actiongroups' with config/sg/sg_action_groups.yml 
   SUCC: Configuration for 'actiongroups' created or updated
Invalid license
Invalid license
Invalid license
Invalid license
Invalid license
Done with failures

```

sg_config file:

```
searchguard:
  dynamic:
    http:
      xff:
        enabled: false
    authc:
      basic_internal_auth_domain:
        http_authenticator:
          type: basic
        authentication_backend:
          type: intern

```

**Comment by floragunncom** [comment]
We are aware of this, fix is in the making ....

**Comment by jochenkressin** [comment]
Workaround for the moment is to add the license string also to the sg_config.yml files like:

```
searchguard:
  dynamic:
    license: |-
      LS0tLS1CRUdJTiBQR1AgU0lHTkVEIE1FU
      1NBR0UtLS0tLQpIYXNoOiBTSEE1M
      ...    
    http:
      ...
    authc:          
      ...
 ```

as described here: https://docs.search-guard.com/latest/search-guard-enterprise-edition

**Comment by iahmad-khan** [comment]
It will not remove the searchguard configuration done through web
interface? Currently it's not only reporting that error but also removed
all the other configuration done through kibana web interface

**Comment by iahmad-khan** [comment]
Actually putting the license in the sg_config results in the same error. 

Some tasks are easier to do in kiabana , but other tasks are to be done in yaml files , [ i think not everything is available in kibana interface] , so when I run the sgadmin, it should not touch what is done by kiabna interface , and it should only update what I have updated in yaml files.  

But this is very strange that it even removes the license that was put using kibana interface, and because i was getting errors put license using the yaml file.

**Comment by floragunncom** [comment]
@iahmad-khan Lets focus on the license handling here in this issue.What do you mean with "Actually putting the license in the sg_config results in the same error."

As Jochen said: You can use the -f option in sgadmin to upload only a specific config file or you use the -r option to retrieve the current config, edit them in a yml editor and upload again. That would be the recommended approach because it will be not very likely that we introduce a sophisticated detection/merge mechanism when which config part was touched by the UI or by sgadmin.

**Comment by floragunncom** [comment]
https://github.com/floragunncom/search-guard/commit/dba296587fb1d828d70295478dbf0736492149ed#diff-14c2c2c1797c20ecdcf1e90a4c41fa3b

**Comment by iahmad-khan** [comment]
I followed the searchguard docs to put the license in yaml but its not pushing the license , maybe I am making some some syntax error. 

But why its removing the license that is already there that i have put using kibana interface  , not all searchguard configuration is possible using kibana. So when you do something from sg_config file , and run sg_admin , it removes everything that was done using kibana. ideally it should not remove things that were done using kibana.

**Comment by iahmad-khan** [comment]
Can you please make a fix in next release of searuchguard , not to change the configuration done by kibana interface of searchguard when we run sg_admin from command line for something that is not available in kibana interface?

**Comment by floragunncom** [comment]
That is not possible. The configuration is stored inside an Elasticsearch index and does not get written back to configuration files automatically. That is the point of this approach, not having to place any config files on each of your nodes in the cluster. And as I said, it will be not very likely that we introduce a sophisticated detection/merge mechanism regarding which config part was touched by the UI or by sgadmin. So the recommended way is to retrieve the config first with the -r switch before making changes.

**Comment by iahmad-khan** [comment]
Ok, I just tried with -r , it works fine , except i have to rename the files that were retrived. The desired bahavior of -r should be to backup the existing files with some other names and bring the latest ones from cluster with those original names that are recognized by sg_admin by default.

**Comment by jochenkressin** [comment]
Fixed in Search Guard 22.3 which has just been released. Valid licenses do not get overwritten anymore:

https://docs.search-guard.com/latest/changelog-searchguard-6-x-22_3
